### PR TITLE
[CodeCompletion] Migrate ForEachSequence and PostfixExprBeginning to solver-based

### PR DESCRIPTION
**Pull request by ahoppen**
Migrate two more completion kinds (ForEachSequence and PostfixExprBeginning) to the solver-based implementation.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: lib/IDE/TypeCheckCompletionCallback.cpp [hunk 1]
```diff
@@ -92,18 +93,93 @@ Type swift::ide::getTypeForCompletion(const constraints::Solution &S, Expr *E) {
...
+///     Swift.(file).Optional extension.~=])
+///   (argument_list implicit
+///     (argument
+///       (code_completion_expr implicit type='$T1'))
+///     (argument
+///       (declref_expr implicit decl=swift_ide_test.(file).foo(x:).$match))))
+/// \endcode
+/// If the code completion expression occurs in such an AST, return the
+/// declaration of the \c $match variable, otherwise return \c nullptr.
+static VarDecl *getMatchVarIfInPatternMatch(Expr *E, ConstraintSystem &CS) {
```
Comment (xedin) [inline lib/IDE/TypeCheckCompletionCallback.cpp:123]: Good news is that `ConstraintSystem` already has a way to check this `ConstraintSystem::isPatternMatchingOperator`

**Review by xedin — APPROVED** [review]

File: lib/IDE/TypeCheckCompletionCallback.cpp (hunk 1 above)
Comment (ahoppen) [reply lib/IDE/TypeCheckCompletionCallback.cpp:123]: Nice. Thanks for catching this. Surely removes some ugly code.

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Pull request merged**